svc.go

```diff
@@ -48,9 +48,10 @@ var (
 
 // AutoNATService provides NAT autodetection services to other peers
 type AutoNATService struct {
-	ctx    context.Context
-	h      host.Host
-	dialer host.Host
+	ctx         context.Context
+	h           host.Host
+	dialer      network.Dialer
```

Comment:
These interfaces are a bit wonky and need to be refactored. Let's just use a host.

Reply:
(or `network.Network`)

Reply:
went with just a `network.Dialer`, since the dialer interface includes `.Peerstore`.

Reply:
`network.Network` embeds the `network.Dialer` interface.